lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureCloudBlockBlob.cs

```diff
@@ -54,70 +57,71 @@ public CopyState CopyState
             }
         }
 
-        public AzureCloudBlockBlob(Uri uri, StorageCredentials credentials)
+        public AzureCloudBlockBlob(Uri uri, StorageCredentials credentials, BlobRequestOptions requestOptions)
         {
             _blob = new CloudBlockBlob(uri, credentials);
+            _requestOptions = requestOptions;
         }
 
         public Stream Open()
         {
             #if REEF_DOTNET_BUILD
-                var task = _blob.OpenReadAsync();
+                var task = _blob.OpenReadAsync(null, _requestOptions, null);
                 task.Wait();
                 return task.Result;
             #else
-                return _blob.OpenRead();
+                return _blob.OpenRead(null, _requestOptions, null);
             #endif
         }
 
         public Stream Create()
         {
             #if REEF_DOTNET_BUILD
-                var task = _blob.OpenWriteAsync();
+                var task = _blob.OpenWriteAsync(null, _requestOptions, null);
                 task.Wait();
                 return task.Result;
             #else
-                return _blob.OpenWrite();
+                return _blob.OpenWrite(null, _requestOptions, null);
             #endif
         }
 
         public bool Exists()
         {
-            var task = _blob.ExistsAsync();
+            var task = _blob.ExistsAsync(_requestOptions, null);
             task.Wait();
             return task.Result;
```

Comment:
here and everywhere: same as above. I think we don't need an explicit `task.Wait()` call if it is followed by `task.Result` (which will block until the task is completed anyway)

Reply:
removed

---
In reply to: [197577058](https://github.com/apache/reef/pull/1470#discussion_r197577058) [](ancestors = 197577058)